pyaerocom/io/mscw_ctm/reader.py

```diff
@@ -421,15 +468,15 @@ def _check_files_in_data_dir(self, data_dir):
             list of file matches
 
         """
-        files = []
-        for freq in self.FREQ_CODES.keys():
-            files.append(self.FILE_FREQ_TEMPLATE.format(freq=freq))
+        files: list[str] = [
+            x for x in os.listdir(data_dir) if os.path.isfile(os.path.join(data_dir, x))
```

Comment:
Consider using pathlib https://docs.python.org/3/library/pathlib.html to avoid os.path.join(), e.g.
```python
from pathlib import Path
p = Path(data_dir)
files: list[Path] = [x for x in p.iterdir() if x.is_file()]
```

Not sure how often the functions ` _check_files_in_data_dir` and `_get_files_from_folders` are called. If they are called several times, consider caching `listdir`/`iterdir` since huge directories might be slow.